impl/src/fmt/debug.rs

```diff
@@ -362,4 +393,94 @@ impl<'a> Expansion<'a> {
             })
         }
     }
+
+    fn contains_type_variable(&self, ty: &syn::Type) -> bool {
+        match ty {
+            syn::Type::Path(syn::TypePath { qself, path }) => {
+                if let Some(qself) = qself {
+                    if self.contains_type_variable(&qself.ty) {
+                        return true;
+                    }
+                }
+
+                if let Some(ident) = path.get_ident() {
+                    self.type_variables
+                        .iter()
+                        .any(|type_var| *type_var == ident)
+                } else {
+                    path.segments.iter().any(|segment| {
+                        self.type_variables
+                            .iter()
+                            .any(|type_var| *type_var == &segment.ident)
```

Comment:
This is outside my Rust knowledge, but I'm starting to think this is wrong. If the type is a path (like `std::vec::Vec`), then it cannot contain a type variable, correct? We only care if it is the only segment or itself a type variable (`T` in `std::vec::Vec<T>`) at some part of the path.